src/parser/monk/mistweaver/modules/talents/AverageTimeBetweenRSKs.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+
+import TalentStatisticBox from 'interface/others/TalentStatisticBox';
+import { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import Events from 'parser/core/Events';
+
+/* Add in Statistic box to show average time between RSK casts when Rising Mist is talented.
+*/
+
+class TimeBetweenRSKs extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+  };
+
+  totalRSKCasts = 0;
+  rskEvents = [];
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.RISING_MIST_TALENT.id);
+    if(!this.active){
+      return;
+    }
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.RISING_SUN_KICK), this.onRSK);
+  }
+
+  onRSK(event) {
+    const spellId = event.ability.guid;
+    if (SPELLS.RISING_SUN_KICK.id !== spellId) {
+      return;
+    }
+    this.totalRSKCasts += 1;
+    this.rskEvents.push(event.timestamp);
+  }
+
+  get lastRSKTimestamp() {
+    return this.rskEvents[this.totalRSKCasts-1];
```

Comment:
If you only every care about the last rsk you don't need to save all of them. You could just save 1 and keep over riding it till the end of the fight.